src/main/java/core/basesyntax/handler/impl/SupplyOperationHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.handler.impl;
+
+import core.basesyntax.handler.OperationHandler;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.storage.Storage;
+
+public class SupplyOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        Storage.addFruit(transaction.getFruit(), transaction.getQuantity());
```

Comment:
The 'addFruit' method should be named according to its function. Since it's modifying the quantity of the fruit in the storage, a name like 'updateFruitQuantity' would be more descriptive and align with the checklist's recommendation for clear method naming.